File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/ReCaptchaCallbackTest.java
/*
 * Copyright (c) 2019 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.android.auth.callback;

import android.content.Context;

import androidx.test.core.app.ApplicationProvider;

import org.forgerock.android.auth.FRListenerFuture;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.robolectric.RobolectricTestRunner;

import java.util.concurrent.ExecutionException;

import static org.junit.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class ReCaptchaCallbackTest {

    @Test
    public void basicTest() throws JSONException {
        JSONObject raw = new JSONObject(" {\n" +
                "            \"type\": \"ReCaptchaCallback\",\n" +
                "            \"output\": [\n" +
                "                {\n" +
                "                    \"name\": \"recaptchaSiteKey\",\n" +
                "                    \"value\": \"6Lf3tbYUAAAAAEm78fAOFRKb-n1M67FDtmpczIBK\"\n" +
                "                }\n" +
                "            ],\n" +
                "            \"input\": [\n" +
                "                {\n" +
                "                    \"name\": \"IDToken1\",\n" +
                "                    \"value\": \"\"\n" +
                "                }\n" +
                "            ]\n" +
                "        }");
        ReCaptchaCallback reCaptchaCallback = new ReCaptchaCallback(raw, 0);

        assertEquals("6Lf3tbYUAAAAAEm78fAOFRKb-n1M67FDtmpczIBK", reCaptchaCallback.getReCaptchaSiteKey());

    }
}

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/ReCaptchaCallbackTest.kt
/*
 * Copyright (c) 2023 - 2025 Ping Identity Corporation. All rights reserved.
 *
 *  This software may be modified and distributed under the terms
 *  of the MIT license. See the LICENSE file for details.
 */
package org.forgerock.android.auth.callback

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class ReCaptchaCallbackTest {
    @Test
    @Throws(JSONException::class)
    fun basicTest() {
        val raw = JSONObject(""" {
            "type": "ReCaptchaCallback",
            "output": [
                {
                    "name": "recaptchaSiteKey",
                    "value": "6Lf3tbYUAAAAAEm78fAOFRKb-n1M67FDtmpczIBK"
                }
            ],
            "input": [
                {
                    "name": "IDToken1",
                    "value": ""
                }
            ]
        }""")
        val reCaptchaCallback = ReCaptchaCallback(raw, 0)
        Assert.assertEquals("6Lf3tbYUAAAAAEm78fAOFRKb-n1M67FDtmpczIBK",
            reCaptchaCallback.reCaptchaSiteKey)
    }

    @Test
    @Throws(JSONException::class)
    fun testCustomCaptchaCallback() {
        val raw = JSONObject(""" {
            "type": "ReCaptchaCallback",
            "output": [
                {
                    "name": "recaptchaSiteKey",
                    "value": "6Lf3tbYUAAAAAEm78fAOFRKb-n1M67FDtmpczIBK"
                }
            ],
            "input": [
                {
                    "name": "IDToken1",
                    "value": ""
                }
            ]
        }""")
        val reCaptchaCallback = ReCaptchaCallback(raw, 0)
        val expectedValue = "custom_captcha_token"
        Assert.assertEquals("6Lf3tbYUAAAAAEm78fAOFRKb-n1M67FDtmpczIBK",
            reCaptchaCallback.reCaptchaSiteKey)
        reCaptchaCallback.setValue("custom_captcha_token")
        val jsonArray: JSONArray = reCaptchaCallback.contentAsJson.get("input") as JSONArray
        val inputValue = (jsonArray[0] as JSONObject).get("value")
        Assert.assertEquals(expectedValue,
            inputValue)
    }
}